app/scripts/metamask-controller.js

```diff
@@ -5164,14 +5160,18 @@ export default class MetamaskController extends EventEmitter {
   }
 
   /**
-   * Gets the permitted accounts for the specified origin. Returns an empty
-   * array if no accounts are permitted.
+   * Gets the sorted permitted accounts for the specified origin. Returns an empty
+   * array if no accounts are permitted or the wallet is locked. Returns any permitted
+   * accounts if the wallet is locked and `ignoreLock` is true. This lock bypass is needed
+   * for the eth_requestAccounts and wallet_getPermission handlers which respectively either
+   * handle locked wallet behavior internally or ignore it entirely.
```

Comment:
Sorry for more nitpicking just feel like this is still confusing...
```suggestion
   * Gets the sorted permitted accounts for the specified origin. Returns an empty
   * array if no accounts are permitted or the wallet is locked. Returns any permitted
   * accounts if the wallet is locked and `ignoreLock` is true. This lock bypass is needed
   * for the `eth_requestAccounts` & `wallet_getPermission` handlers both of which  
   * return permissioned accounts to the dapp when the wallet is locked.
   * TODO: we should also return permissioned accounts on `eth_accounts` calls 
   * when the wallet is locked
```